src/fu-main.c

```diff
@@ -43,8 +44,21 @@ typedef struct {
 	PolkitAuthority		*authority;
 	guint			 owner_id;
 	FuEngine		*engine;
+	gboolean		 update_in_progress;
+	gboolean		 pending_sigterm;
 } FuMainPrivate;
 
+static gboolean
+fu_main_sigterm_cb (gpointer user_data)
+{
+	FuMainPrivate *priv = (FuMainPrivate *) user_data;
+	if (!priv->update_in_progress)
+		return G_SOURCE_REMOVE;
+	g_warning ("Received SIGTERM during a firmware update, ignoring");
+	priv->pending_sigterm = TRUE;
+	return G_SOURCE_CONTINUE;
```

Comment:
I'm a bit confused whay CONTINUE and REMOVE mean in this context; is it that CONTINUE will continue listening for SIGTERM, and REMOVE stops listening? I'm wondering if an update is not happening, do we remove the source without actually quitting the loop?

Reply:
I realize I didn't actually check that specific scenario.  After checking, you're definitely correct.  It exits the handler and you need 2 `SIGTERM`s to get it to quit.

I've adjusted to quit the main loop in that scenario and confirmed it quits after one `SIGTERM`.  As a benefit this should also let the daemon clean up more nicely when stopping the systemd unit.